core/src/plugins/filed/python/vmware/BareosFdPluginVMware.py

```diff
@@ -328,20 +330,81 @@ def check_plugin_options(self, mandatory_options=None):
+                    % self.options["snapshot_retries"],
+                )
+                return bareosfd.bRC_Error
+
+        if self.options.get("snapshot_retry_wait"):
+            try:
+                self.vadp.snapshot_retry_wait = int(self.options["snapshot_retry_wait"])
+            except ValueError:
+                bareosfd.JobMessage(
+                    bareosfd.M_FATAL,
+                    "Invalid value '%s' for snapshot_retry_wait, only digits allowed\n"
+                    % self.options["snapshot_retry_wait"],
+                )
+                return bareosfd.bRC_Error
+
+            if self.vadp.snapshot_retry_wait < 0:
+                bareosfd.JobMessage(
+                    bareosfd.M_FATAL,
+                    "Invalid value '%s' for snapshot_retry_wait, negative values not allowed\n"
+                    % self.options["snapshot_retry_wait"],
+                )
+                return bareosfd.bRC_Error
+
         for options in self.options:
             bareosfd.DebugMessage(
                 100,
                 "Using Option %s=%s\n"
                 % (option, StringCodec.encode(self.options[option])),
             )
 
-        if (
-            not self.options.get("localvmdk") == "yes"
-            and "vcthumbprint" not in self.options
-        ):
-            # if vcthumbprint is not given in options, retrieve it
-            if not self.vadp.retrieve_vcthumbprint():
-                return bareosfd.bRC_Error
+        for options in self.options:
+            bareosfd.DebugMessage(
+                100,
+                "Using Option %s=%s\n"
+                % (option, StringCodec.encode(self.options[option])),
+            )
+
+        if not self.options.get("localvmdk") == "yes":
+            if "vcthumbprint" in self.options:
+                # fetch the thumbprint and compare it with the passed thumbprint,
+                # warn if they don't match as VixDiskLib will return "unknown error"
+                # when running bareos-vadp-dumper
+                self.vadp.fetch_vcthumbprint()
+                if self.options["vcthumbprint"] != self.vadp.fetched_vcthumbprint:
+                    bareosfd.JobMessage(
+                        bareosfd.M_ERROR,
+                        "The configured vcthumbprint %s does not match the server thumbprint %s, "
+                        "check and update your config! Otherwise bareos_vadp_dumper will get the "
+                        "error message 'Unknown error' from the API\n"
+                        % (
+                            self.options["vcthumbprint"],
+                            self.vadp.fetched_vcthumbprint,
+                        ),
+                    )
+
+            else:
+                # if vcthumbprint is not given in options, retrieve it
+                if not self.vadp.retrieve_vcthumbprint():
+                    return bareosfd.bRC_Error
```

Comment:
Is the automatically retrieved thumbprint of interest? Would it make sense to also put it into the joblog?